src/initialization/InitMeshRefinement.cpp

```diff
@@ -116,13 +113,42 @@ namespace impactx
         if (x_min == x_max || y_min == y_max || z_min == z_max)
             throw std::runtime_error("Flat beam detected. This is not yet supported: https://github.com/ECP-WarpX/impactx/issues/44");
 
+        amrex::ParmParse pp_geometry("geometry");
+        bool dynamic_size = true;
+        pp_geometry.query("dynamic_size", dynamic_size);
+
+        amrex::RealBox rb;
+        if (dynamic_size)
+        {
+            // The box is expanded beyond the min and max of particles.
+            // This controlled by the variable `frac` below.
+            amrex::Real frac = 0.1;
+            pp_geometry.query("prob_relative", frac);
+
+            rb = {
+                {x_min - frac * (x_max - x_min), y_min - frac * (y_max - y_min),
+                 z_min - frac * (z_max - z_min)}, // Low bound
+                {x_max + frac * (x_max - x_min), y_max + frac * (y_max - y_min),
+                 z_max + frac * (z_max - z_min)}  // High bound
+            };
+        }
+        else
+        {
+            amrex::Vector<amrex::Real> prob_lo;
```

Comment:
Is this needed here?
It seems that, when `dynamic_size` is `false`, the code will still call `amrex::Geometry::ResetDefaultProbDomain` and `amrex::AmrMesh::SetGeometry` (even though `prob_lo` and `prob_hi` are not changing.)

Would it make sense to instead call `return` early in this function if `dynamic_size` is `false`?

Reply:
Yes, that is intentional in case someone changed from dynamic to static size after part of a simulation.

`amrex::Geometry::ResetDefaultProbDomain` and `amrex::AmrMesh::SetGeometry` are cheap functions.

Reply:
```suggestion
            // note: we read and set the size again because an
            //       interactive/Python user might have changed it between steps
            amrex::Vector<amrex::Real> prob_lo;
```